docs/manual/src/udl/ext_types_wrapped.md

```diff
@@ -1,29 +1,55 @@
-# External types which wrap primitive types
+# External types which wrap UDL types
 
-*Note: The facility described in this document is not yet available for all foreign language
-bindings.*
+*Note: The facility described in this document is not yet available for the Ruby bindings.*
 
-It's possible to expose types in your Rust crate to UniFFI, so long as the type can
-be expressed as one of UniFFI's [builtin_types](./builtin_types.md).
+`Wrapped` types allow you to extend the UniFFI type system with custom types.
+These types rely on a "normal" UDL type move data across the FFI, then use a
+custom type to wrap those types on the scaffolding and/or bindings side.
 
-For example, consider the following trivial Rust abstraction for a "handle" which wraps an
+Some examples:
+  - You have Rust type named `Handle` which uses the new-type pattern to wrap a
+    string and provides additional functionality.  We can have `Handle` wrap
+    `String` on the scaffolding side so that the Rust code gets to use that
+    functionality, but on the bindings side the type behaves like a normal
+    string.
+  - You've updated a `Record` in your Rust library, but your bindings code is
+    still using the legacy version of it.  You can wrap the `Record` on the
+    bindings side with a type that provides the legacy API.
+  - You want to use `Json` to exchange data between your Rust library and it's
```

Comment:
"both sides with a `JsonObject`" isn't clear to me - I think I know what you mean, but that's just because we've spent months talking about this :) Specifically, `JsonObject` has no meaning to Python nor to Rust. I think generalizing this might help - something like "The string can be wrapped with whatever type makes sense on both sides -for example, the Rust side might use a `serde_json::Value` enum whereas your [Kotlin code might use a {whatever}|Python code might use objects returned by the `json` module]"